src/core/pyproxy.js

```diff
@@ -188,6 +194,46 @@ Module.getPyProxyClass = function (flags) {
 
 // Static methods
 Module.PyProxy_getPtr = _getPtr;
+Module.pyproxy_mark_borrowed = function (proxy) {
+  proxy.$$.borrowed = true;
+};
+
+const pyproxy_cache_destroyed_msg =
+  "This borrowed attribute proxy was automatically destroyed in the " +
+  "process of destroying the proxy it was borrowed from. Try using the 'copy' method.";
+
+function pyproxy_destroy_cache(cache) {
```

Comment:
Maybe we could add a doc string with a sentence explaining what this function does. It's non obvious that despite its name it does not always destroy the cache (depending on the refcnt value). Maybe a better name would be `pyproxy_decref_cache` ?

Reply:
Yeah I think I named it before I figured out that the cache would need to be refcounted.